src/widgets/block.rs

```diff
@@ -221,38 +335,81 @@ impl<'a> Block<'a> {
         self.title_position(Position::Bottom)
     }
 
-    /// Positions all titles that don't have a position
+    /// Positions all [titles](Title) that don't have a [`Position`].
+    ///
+    /// This effectively acts as a default for titles that don't explicitly set an [`Position`].
+    ///
     /// # Example
+    ///
     /// This example position all titles on the bottom except "top" title
     /// ```
     /// # use ratatui::widgets::{Block, BorderType, Borders, block::title::{Position, Title}};
     /// Block::default()
-    ///   // This title won't be aligned in the center
-    ///   .title(Title::from("top").position(Position::Top))
-    ///   .title("foo")
-    ///   .title("bar")
-    ///   .title_position(Position::Bottom);
+    ///     // This title won't be aligned in the center
+    ///     .title(Title::from("top").position(Position::Top))
+    ///     .title("foo")
+    ///     .title("bar")
+    ///     .title_position(Position::Bottom);
     /// ```
     pub const fn title_position(mut self, position: Position) -> Block<'a> {
         self.titles_position = position;
         self
     }
 
+    /// Defines the style of the borders.
+    ///
+    /// If a [`Block::style`] is defined, `border_style` will be applied on top of it.
+    ///
+    /// # Example
+    ///
+    /// This example shows a `Block` with blue borders.
+    /// ```
+    /// # use ratatui::prelude::*;
+    /// # use ratatui::widgets::{Block, Borders};
+    /// Block::default()
+    ///     .borders(Borders::ALL)
+    ///     .border_style(Style::new().blue());
+    /// ```
     pub const fn border_style(mut self, style: Style) -> Block<'a> {
         self.border_style = style;
         self
     }
 
+    /// Defines the block style.
+    ///
+    /// This is the most generic [`Style`] a block can receive, it will be merged with any other
+    /// more specific style. Elements can be styled further with [`Block::title_style`] and
+    /// [`Block::border_style`].
     pub const fn style(mut self, style: Style) -> Block<'a> {
         self.style = style;
         self
     }
 
+    /// Defines which borders to display.
+    ///
+    /// [`Borders`] can also be styled with [`Block::border_style`] and [`Block::border_type`].
+    ///
+    /// # Examples
+    ///
+    /// Simply show all borders.
+    /// ```
+    /// # use ratatui::widgets::{Borders, Block};
+    /// Block::default().borders(Borders::ALL);
+    /// ```
+    ///
+    /// Display left and right borders.
+    /// ```
+    /// # use ratatui::widgets::{Borders, Block};
+    /// Block::default().borders(Borders::LEFT | Borders::RIGHT);
+    /// ```
     pub const fn borders(mut self, flag: Borders) -> Block<'a> {
         self.borders = flag;
         self
     }
 
+    /// Defines the type of border to use.
+    ///
+    /// See [`BorderType`] for more information.
```

Comment:
This is a little ambiguous (but that flows from the name itself) and makes it hard for a reader to search the docs for how to set this.
Perhaps "Sets the symbols used to display the border (e.g. single line, double line, thick or rounded borders)"?

Reply:
Yes that's clearer with this phrasing.

I kept the reference to `BorderType`, but rephrased it, as this is an explicit list of all border types.